src/gui/addnewtorrentdialog.cpp

```diff
@@ -148,10 +148,9 @@ AddNewTorrentDialog::AddNewTorrentDialog(const BitTorrent::AddTorrentParams &inP
 AddNewTorrentDialog::~AddNewTorrentDialog()
 {
     saveState();
+
+    delete m_contentDelegate;
```

Comment:
because `m_contentModel = new TorrentContentFilterModel(this);`
the parent pointer is setup correctly.
And I don't see dependency between delegate & model in code.